### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable when running stream_data()

### ISSUE THREAD

**Issue by DerekChia** [yhilpisch/tpqoa#1]
Got this error when calling `oanda.stream_data('EUR_USD', stop=7)`. Is there something wrong with v20? Thanks.

> Traceback (most recent call last):
>   File "o.py", line 10, in <module>
>     oanda.stream_data('EUR_USD')
>   File "/anaconda3/envs/py36/lib/python3.6/site-packages/tpqoa-0.0.43-py3.6.egg/tpqoa/tpqoa.py", line 206, in stream_data
> TypeError: 'NoneType' object is not subscriptable

**Comment by yhilpisch** [comment]
Hart to tell with this snippet alone. Please read the following and maybe rephrase/complete the code/problem.

How to create a Minimal, Complete, and Verifiable example
https://stackoverflow.com/help/mcve

**Comment by DerekChia** [comment]
Just found out that I was using an old v1 API key and hence the error. My bad.

**Comment by anguslou** [comment]
> Just found out that I was using an old v1 API key and hence the error. My bad.

Do you need to uninstall v1 API to install v2 API?
I also installed v1 and then v2, still have the error "NoneType" object is not Subscriptable. Not sure what is still missing.

**Comment by yhilpisch** [comment]
The v1 API is long deprecated. Only install `tpqoa` via:

`pip install git+https://github.com/yhilpisch/tpqoa.git`

**Comment by anguslou** [comment]
> The v1 API is long deprecated. Only install `tpqoa` via:
> 
> `pip install git+https://github.com/yhilpisch/tpqoa.git`

I have installed v2 but the NoneType error is not fixed.

**Comment by yhilpisch** [comment]
Without exact details, it is hard to tell. Maybe your key is not valid or something else.

**Comment by miali88** [comment]
I've installed tpqoa using the link provided above, and I'm connected to the api, able to get_history, to put buy/sell orders through no problem. But the stream_data function brings the same hiccup.

```
[1] import tpqoa
[2] api=tpqoa.tpqoa('pyalgo.cfg')
[3] instrument ='EUR_USD'
[4] api.stream_data(instrument, stop=10)

```

With the error:

> TypeError                                 Traceback (most recent call last)
> ~\AppData\Local\Temp/ipykernel_1752/2279309372.py in <module>
> ----> 1 api.stream_data(instrument, stop=10)
> ~\miniconda3\envs\env-pyalgo\lib\site-packages\tpqoa-0.0.55-py3.9.egg\tpqoa\tpqoa.py in stream_data(self, instrument, stop, ret, callback) 
>     356                 else:
>     357                     self.on_success(msg.time,
> --> 358                                     float(msg.bids[0].dict()['price']),
>     359                                     float(msg.asks[0].dict()['price']))
>     360                 if stop is not None:
> 
> TypeError: 'NoneType' object is not subscriptable

​

**Comment by yhilpisch** [comment]
At what day and time did you execute the streaming code? E.g. on Sundays it does not work.

For me it all works as desired:

```
import tpqoa
api = tpqoa.tpqoa('oanda.cfg')
api.stream_data('EUR_USD', stop=5)
2021-09-21T09:00:12.741486747Z 1.17336 1.17349
2021-09-21T09:00:18.466550104Z 1.17337 1.17351
2021-09-21T09:00:20.327798203Z 1.1734 1.17353
2021-09-21T09:00:20.600701008Z 1.17339 1.17351
2021-09-21T09:00:20.931842223Z 1.1734 1.17352
```

**Comment by miali88** [comment]
Tried Tuesday 3am BST, and again at 5pm BTS. Same error, after restarting machine.

**Comment by miali88** [comment]
Tried Tuesday 3am BST, and again at 5pm BTS. Same error, after restarting machine.
